compiler/passes/normalize.cpp

```diff
@@ -1873,14 +1883,15 @@ static void normVarTypeInference(DefExpr* defExpr) {
         argExpr->baseExpr->replace(new UnresolvedSymExpr("init"));
 
         // Add _mt and _this (insert at head in reverse order)
-        argExpr->insertAtHead(var);
+        argExpr->insertAtHead(new NamedExpr("this", new SymExpr(var)));
```

Comment:
Would you prefer that I only add this for generic records?  I would be happy to perform that limitation, but I do believe it is necessary for them